### PR TITLE
Dev/alha

### PR DESCRIPTION
**Pull request by alexjhawk**

### CODE REVIEW

File: src/main/java/com/hms_networks/sc/cumulocity/config/CConnectorConfigFile.java [hunk 1]
```diff
@@ -368,6 +381,25 @@ public boolean getCumulocitySubscribeToErrors() {
...
+   *
+   * @return Cumulocity port setting
+   */
+  public String getCumulocityPort() {
+    String port = PORT_DEFAULT;
+    try {
+      port =
+          configurationObject
+              .getJSONObject(CONFIG_FILE_CUMULOCITY_KEY)
+              .getString(CONFIG_FILE_PORT_KEY);
```
Comment (it-hms) [inline src/main/java/com/hms_networks/sc/cumulocity/config/CConnectorConfigFile.java:395]: I'm confused/concerned about this being a string. Does this mean the configuration file should like this? 
```JSON
{
    "Port": "1883"
}
```
I'm concerned that the following is what users will think, and the logs won't point out the error 
```JSON
{
  "Port": 1883
}
```
I would also be concerned about validating the port..  could you please test 
```JSON
{
  "Port": "shark"
}
```

**Review by it-hms — COMMENTED** [review]
I'm a little concerned about port being a string (not an int) and the validation that we do.  Could you please test some common error scenarios and make sure that the logs reflect the user error.

File: src/main/java/com/hms_networks/sc/cumulocity/config/CConnectorConfigFile.java (hunk 1 above)
Comment (alexjhawk) [reply src/main/java/com/hms_networks/sc/cumulocity/config/CConnectorConfigFile.java:395]: Will guard the value to ensure that it is a valid integer.

The Ewon MQTT port set method actually requires a string itself, thus the string data type is carried throughout here. Converting to an int and back for the configuration field would be a waste of computing time and introduces unneeded complexity.

Comment (alexjhawk) [reply src/main/java/com/hms_networks/sc/cumulocity/config/CConnectorConfigFile.java:395]: Changed - if the user includes a value which is not a valid number, the connector will catch and warn:

*Port does not contain a valid port number. The port must be a valid number between 0 and 65535, enclosed in a string. Using default: 8883*

**Comment by alexjhawk** [comment]
> I'm a little concerned about port being a string (not an int) and the validation that we do. Could you please test some common error scenarios and make sure that the logs reflect the user error.

Added error handling to ensure that the value is a parseable integer. If it is not, an error is shown and the default port of 8883 is used. Values like "shark" or "8883a" will be caught and result in the warning message and default port usage.

File: README.md [hunk 2]
```diff
@@ -114,6 +123,10 @@ This section contains configuration fields which are used to configure the conne
 Parameter to specify the host name of the Cumulocity server. This must be configured prior to
 running the connector.
 
+#### Port (Port)
+
+Optional parameter to specify the port of the Cumulocity server.
```
Comment (it-hms) [inline README.md:128]: Could you please explicitly state that this parameter should be JSON string, and provide an example

**Review by it-hms — APPROVED** [review]
I approve, but please add example port selection in the README so that it is clear that port should be a JSON string and not number

File: README.md (hunk 2 above)
Comment (alexjhawk) [reply README.md]: Updated to say "string parameter" and include the default value of "8883"

**Review by it-hms — APPROVED** [review]

**Pull request merged**